### PR TITLE
Ellipsify long dashboard filter values (strings only)

### LINKED ISSUE #46189: Dashboard filters with long values go off-screen
### Describe the bug

Dashboard string filters with long enough values go off-screen

### To Reproduce

1. Open a dashboard with at least one question
2. Add a string filter (don't forget to link it to a column so the filter is visible after leaving editing mode)
3. When editing the string filter, set pick "Input box" in the "How should people filter on this column?" section of the filter sidebar
4. Save the dashboard
5. Add a long enough value (could be a few sentences)
6. Notice the filter goes off-screen

### Expected behavior

String filters should have a fixed length (~30 characters), ellipsify text, and display the full value in a tooltip on hover

### Logs

_No response_

### Information about your Metabase installation

```JSON
`master`, but it's been like that for at least a few versions
```

### Severity

P2

### Additional context

  
before

### PR DESCRIPTION
**Pull request by kulyk**
Closes #46189

Sets a max width for string filter widgets so they don't go off-screen if a value is long enough

### To verify

1. Open a dashboard with at least one question
2. Add a string filter (don't forget to link it to a column so the filter is visible after leaving editing mode)
3. When editing the string filter, set pick "Input box" in the "How should people filter on this column?" section of the filter sidebar
4. Save the dashboard
5. Add a long enough value (could be a few sentences)
6. Ensure the filter component doesn't go off-screen and that it ellipsifies its contents. Ensure you can hover the filter and see the full value in a tooltip
7. Ensure filters look as expected with shorter values

### Demo

**Before**

before

**After**

after

### CODE REVIEW

File: frontend/src/metabase/parameters/components/ParameterValueWidget.tsx [hunk 1]
```diff
@@ -225,11 +228,17 @@ export const ParameterValueWidget = ({
                   size={16}
                 />
               )}
-              <div className={cx(CS.mr1, CS.textNoWrap)}>
+              <div
+                className={cx(CS.mr1)}
+                style={
+                  isStringParameter(parameter) ? { maxWidth: "190px" } : {}
+                }
+              >
```
Comment (kulyk) [inline frontend/src/metabase/parameters/components/ParameterValueWidget.tsx:236]: Limiting this to string parameters so we don't accidentally crop something else, e.g. a data filter value:


CleanShot 2024-07-26 at 15 30 11@2x

File: frontend/src/metabase/parameters/components/FormattedParameterValue/FormattedParameterValue.tsx [hunk 2]
```diff
@@ -33,25 +39,42 @@ function FormattedParameterValue({
...
-      />
+      <Ellipsified
+        showTooltip={!isPopoverOpen}
+        multiline
+        tooltipMaxWidth={hasLongValue ? 450 : undefined}
+      >
+        {renderContent()}
+      </Ellipsified>
     );
   }
```
Comment (kulyk) [inline frontend/src/metabase/parameters/components/FormattedParameterValue/FormattedParameterValue.tsx:75]: Labels shorter that 80 characters look odd when `tooltipMaxWidth` is set:


CleanShot 2024-07-26 at 15 32 02@2x

File: frontend/src/metabase/parameters/components/FormattedParameterValue/FormattedParameterValue.tsx [hunk 3]
```diff
@@ -33,25 +39,42 @@ function FormattedParameterValue({
 
   const label = getLabel(displayValue);
 
-  if (
-    isFieldFilterUiParameter(parameter) &&
-    hasFields(parameter) &&
-    !isDateParameter(parameter)
-  ) {
+  const renderContent = () => {
```
Comment (EmmadUsmani) [inline frontend/src/metabase/parameters/components/FormattedParameterValue/FormattedParameterValue.tsx:42]: Nit: you could make this a separate component like `FormattedParameterValueInner` and pass `parameter`, `value`, `label` as props.

**Review by EmmadUsmani — APPROVED** [review]
Overall looks good 👍 Would be nice to add a small unit or e2e test just asserting that the text is truncated when we expect it to be, and it's not when it shouldn't

**Review by mazameli — APPROVED** [review]

**Review by cdeweyx — APPROVED** [review]

**Pull request merged**